api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java

```diff
@@ -107,8 +108,16 @@ public BiometricSubject updateSubjectId(String oldId, String newId) {
 
     @Override
     public List<BiometricMatch> search() {
-        M2SysCaptureResponse capture = scanDoubleFingers();
-        return searchService.search(capture, localBioServerClient);
+        M2SysCaptureResponse fingerScan = scanDoubleFingers();
+        List<BiometricMatch> results = searchService.search(fingerScan, localBioServerClient);
+        if (CollectionUtils.isEmpty(results)) {
+            BiometricMatch nationalResult = searchService.findMostAdequate(fingerScan, nationalBioServerClient);
+            if (nationalResult != null) {
+                registrationService.fetchFromNational(new BiometricSubject(nationalResult.getSubjectId()), fingerScan);
```

Comment:
why is the registrationService doing a search? Registration service should be only responsible for registration

Reply:
Is this the MPI fetch? Can we rename to fetchFromMpiByNationalFpId ?

Reply:
What about naming it fetchFromMpiByNationalBiometricSubject?

Reply:
because we need only subjectId but it's a string

Reply:
NationalFpId is shorter and is the same thing as nationalBiometricSubject -both work for me

Reply:
I think we should use the name with word EMPI not an MPI.

Reply:
registrationService registers fingerprints and fetches patient (using registration core)

Reply:
EMPI is an implementation of the MPI, we can have different mpis

Reply:
Okay, good to know.